cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -386,7 +391,16 @@ static PyObject *register_timed_callback(PyObject *self, PyObject *args)
 
     // Extract the time
     PyObject *pTime = PyTuple_GetItem(args, 0);
-    time_ps = PyLong_AsLongLong(pTime);
+    if (PyLong_Check(pTime)) {
+        time_ps = PyLong_AsUnsignedLongLong(pTime);
```

Comment:
The call `PyLong_AsUnsignedLongLong()` needs error handling, according to https://docs.python.org/3/c-api/long.html#c.PyLong_AsUnsignedLongLong.